sound/soc/soc-pcm.c

```diff
@@ -2109,9 +2139,12 @@ int dpcm_be_dai_trigger(struct snd_soc_pcm_runtime *fe, int stream,
 				spin_unlock_irqrestore(&fe->card->dpcm_lock, flags);
 				continue;
 			}
+			if (be->dpcm[stream].be_start == 1)
+				do_trigger = true;
+			be->dpcm[stream].be_start--;
```

Comment:
how about " be->dpcm[stream].be_start--;  if (!be->dpcm[stream].be_start) do_trigger = true ?

Reply:
I changed the order. I prefer an explicit zero here in case something goes wrong.